internal/fwschemadata/data_default.go

```diff
@@ -21,15 +21,15 @@ import (
 // when configRaw contains a null value at the same path.
 func (d *Data) TransformDefaults(ctx context.Context, configRaw tftypes.Value) diag.Diagnostics {
 	var diags diag.Diagnostics
+	var err error
 
 	configData := Data{
 		Description:    DataDescriptionConfiguration,
 		Schema:         d.Schema,
 		TerraformValue: configRaw,
 	}
 
-	// Errors are handled as richer diag.Diagnostics instead.
-	d.TerraformValue, _ = tftypes.Transform(d.TerraformValue, func(tfTypePath *tftypes.AttributePath, tfTypeValue tftypes.Value) (tftypes.Value, error) {
+	d.TerraformValue, err = tftypes.Transform(d.TerraformValue, func(tfTypePath *tftypes.AttributePath, tfTypeValue tftypes.Value) (tftypes.Value, error) {
```

Comment:
This is awesome ❤️ . I don't want to stretch this PR too far, but do we want to just add similar error handling now to other places we follow a similar pattern with `tftypes.Transform`. Quick search shows:
- `(*Data).NullifyCollectionBlocks`
- `(*Data).ReifyNullCollectionBlocks`

Reply:
Created https://github.com/hashicorp/terraform-plugin-framework/issues/938 for tracking.